Mathlib/Data/Int/WithZero.lean

```diff
@@ -0,0 +1,89 @@
+/-
+Copyright (c) 2024 María Inés de Frutos-Fernández, Filippo A. E. Nuccio. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: María Inés de Frutos-Fernández, Filippo A. E. Nuccio
+-/
+import Mathlib.Data.NNReal.Basic
+
+/-!
+# WithZero
+
+In this file we provide some basic API lemmas for the `WithZero` construction and we define
+the morphism `withZeroMultIntToNNReal`.
+
+## Main Definitions
+
+* `withZeroMultIntToNNReal` : The `MonoidWithZeroHom` from `ℤₘ₀ → ℝ≥0` sending `0 ↦ 0` and
+  `x ↦ e^(Multiplicative.toAdd (WithZero.unzero hx)` when `x ≠ 0`, for a nonzero `e : ℝ≥0`.
+
+## Main Results
+
+* `withZeroMultIntToNNReal_strictMono` : The map `withZeroMultIntToNNReal` is strictly
+   monotone whenever `1 < e`.
+
+## Tags
+
+WithZero, multiplicative, nnreal
+-/
+
+noncomputable section
+
+open scoped NNReal
+
+open Multiplicative WithZero
+
+local notation "ℤₘ₀" => WithZero (Multiplicative ℤ)
```

Comment:
The names of these theorems become pretty long. Would it make sense to make this notation into an `abbrev` and then use the `WithZeroMulInt` namespace. Then the morphism below would become `WithZeroMulInt.toNNReal`.

Reply:
Done!